tests/sort_test.py

```diff
@@ -12,7 +12,33 @@ def testSort(self):
         spda = ak.sort(pda)
         maxIndex = spda.argmax()
         self.assertTrue(maxIndex > 0)
-   
+
+    def testBitBoundaryHardcode(self):
+
+        # test hardcoded 16/17-bit boundaries with and without negative values
+        a = ak.array([1, -1, 32767]) # 16 bit
+        b = ak.array([1,  0, 32768]) # 16 bit
+        c = ak.array([1, -1, 32768]) # 17 bit
+        assert ak.is_sorted(ak.sort(a))
+        assert ak.is_sorted(ak.sort(b))
+        assert ak.is_sorted(ak.sort(c))
+
+        # test hardcoded 64-bit boundaries with and without negative values
+        d = ak.array([1, -1, 2**63-1])
+        e = ak.array([1,  0, 2**63])
```

Comment:
Did you mean `2**63-1` here as well? I don't think `2**63` will transfer over correctly, since it is out of bounds for `int64`.

Reply:
Ah, I did mean `2**63` to get something that's an unsigned 64-bit value, but I was just thinking about this from the perspective of a bag-of-bits, forgetting that the actual types are signed 64-bits so this would overflow. I'll change it.

Reply:
Changed to `2**63-1`